lib/proof/datacid.go

```diff
@@ -0,0 +1,165 @@
+package proof
+
+import (
+	"math/bits"
+	"runtime"
+
+	"github.com/ipfs/go-cid"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-commp-utils/nonffi"
+	"github.com/filecoin-project/go-commp-utils/zerocomm"
+	commcid "github.com/filecoin-project/go-fil-commcid"
+	commp "github.com/filecoin-project/go-fil-commp-hashhash"
+	"github.com/filecoin-project/go-state-types/abi"
+)
+
+type DataCIDSize struct {
+	PayloadSize int64
+	PieceSize   abi.PaddedPieceSize
+	PieceCID    cid.Cid
+}
+
+const commPBufPad = abi.PaddedPieceSize(8 << 20)
+const CommPBuf = abi.UnpaddedPieceSize(commPBufPad - (commPBufPad / 128)) // can't use .Unpadded() for const
+
+type ciderr struct {
+	c   cid.Cid
+	err error
+}
+
+/*
+DataCidWriter is used as follows:
```

Comment:
Add a bit as to why I want to use this:
This allows streaming reads to compute the PieceCID, PayloadSize, and padded piece size